src/tests/Microsoft.PowerFx.Core.Tests/TypesAndDisplayNameTests/LazyTypeDisplayNameTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Core.Tests.TypesAndDisplayNameTests
+{
+    public class LazyTypeDisplayNameTests
+    {        
+        public class TestLazyRecordTypeWithDisplayNames : RecordType
+        {
+            public override IEnumerable<string> FieldNames { get; }
+
+            public TestLazyRecordTypeWithDisplayNames()
+                : base()
+            {
+                FieldNames = new List<string>() { "Foo", "Bar", "Baz" };
+            }
+
+            public override bool TryGetFieldType(string name, out FormulaType type)
+            {
+                type = name switch
+                {
+                    "Foo" => FormulaType.Number,
+                    "Bar" => FormulaType.DateTime,
+                    "Baz" => FormulaType.Boolean,
+                    _ => FormulaType.Blank
+                };
+
+                return type != FormulaType.Blank;
+            }
+
+            public override bool TryGetDisplayName(string logicalName, out string displayName)
+            {                
+                displayName = logicalName switch
+                {
+                    "Foo" => "FooDisplay",
+                    "Bar" => "BarDisplay",
+                    "Baz" => "BazDisplay",
+                    _ => null
+                };
+
+                return displayName != null;
+            }
+
+            public override bool TryGetLogicalName(string displayName, out string logicalName)
```

Comment:
>TryGetLogicalName [](http://example.com/codeflow?start=33&length=17)

As a sanity check -  
can we also emulate TryGetLogicalName via  FieldNames + TryGetDisplayName? 

Put another way, could we have a reasonable default implementation ?

Reply:
Yes, we could iterate FieldNames , and call `TryGetDisplayName` for each, stopping when we find a match. 
Not sure if it's worth having a slow default implementation vs asking hosts to override both.